lib/consensus.js

```diff
@@ -39,6 +41,36 @@ function isValidBlockHeader(newHeader, previousHeaders, network = 'mainnet') {
     && hasGreaterThanMedianTimestamp(newHeader, previousHeaders);
 }
 
+/**
+ * validates an array of transaction objects, an array of tx hashes
+ * or a single transaction object or its hash
+ * against a merkleblock and the local header chain
+ * @param {Object[]|string[]|Object|string} transactions
+ * @param {Object} rawMerkleBlock - a base64 encoded merkleblock
+ * @param {Object} headerChain - an instance of an spvchain
+ * @return {boolean}
+ */
+async function isValidTransaction(transactions, rawMerkleBlock, headerChain) {
+  let tx = [];
+  if (!Array.isArray(transactions)) {
+    if ((typeof transactions === 'string') || (transactions instanceof Transaction)) {
+      tx.push(transactions);
+    } else {
+      throw new Error('Please check type of transactions parameter');
+    }
+  } else {
+    tx = transactions.slice();
+  }
+
+  const merkleBlock = new MerkleBlock(Buffer.from(rawMerkleBlock, 'base64'));
```

Comment:
Why do we accept base64 instead of object or buffer?

Reply:
I thought that is what how the client receives the merkleblocks through the stream? Maybe it would be good for the client to not have to convert it and pass the merkleblocks along as they come in. Which is the correct type we should use?

Reply:
GRPC Web uses base64 internally but in handlers, you are receiving types specified in proto-objects. In this case 'bytes'.

Reply:
done